sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala

```diff
@@ -490,7 +495,12 @@ object OrcUtils extends Logging {
             s"createAggInternalRowFromFooter should not take $x as the aggregate expression")
       }
 
-    val orcValuesDeserializer = new OrcDeserializer(aggSchema, (0 until aggSchema.length).toArray)
-    orcValuesDeserializer.deserializeFromValues(aggORCValues)
+    val orcValuesDeserializer = new OrcDeserializer(schemaWithoutGroupby,
+      (0 until schemaWithoutGroupby.length).toArray)
+    if (aggregation.groupByColumns.length > 0) {
+      new JoinedRow(partitionValues, orcValuesDeserializer.deserializeFromValues(aggORCValues))
```

Comment:
nit: `orcValuesDeserializer.deserializeFromValues(aggORCValues)` can be called only once before the if-else branch.